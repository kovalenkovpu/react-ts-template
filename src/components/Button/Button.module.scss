.button {
  padding: 1rem;
  margin: 2px;
  border-width: 1px;

  &.primary {
    background-color: #8a8af5;
  }

  &.error {
    background-color: #e97b7a;
  }
}
