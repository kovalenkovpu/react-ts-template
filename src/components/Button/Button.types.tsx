enum BUTTON_TYPES {
  PRIMARY = 'primary',
  ERROR = 'error',
}

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  btnType?: BUTTON_TYPES;
}

export { BUTTON_TYPES };
export type { ButtonProps };
