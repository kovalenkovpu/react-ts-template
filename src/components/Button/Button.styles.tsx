import styled from 'styled-components';

import { BUTTON_TYPES, ButtonProps } from './Button.types';

const StyledButton = styled.button<ButtonProps>`
  padding: 1rem;
  margin: 2px;
  background-color: ${props =>
    props.btnType === BUTTON_TYPES.ERROR ? '#e97b7a' : '#8a8af5'};
  border-width: 1px;
`;

export { StyledButton };
