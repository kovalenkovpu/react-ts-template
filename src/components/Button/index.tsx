import * as React from 'react';

import { StyledButton } from './Button.styles';
import { BUTTON_TYPES, ButtonProps } from './Button.types';

const Button: React.FC<ButtonProps> = ({
  btnType = BUTTON_TYPES.PRIMARY,
  children,
  ...rest
}) => {
  return (
    <StyledButton btnType={btnType} {...rest}>
      {children}
    </StyledButton>
  );
};

export { Button };
