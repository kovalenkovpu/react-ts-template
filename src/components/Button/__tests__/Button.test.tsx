import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Button } from '..';
import { BUTTON_TYPES } from '../Button.types';

describe('Button component', () => {
  test('should match the snapshot', () => {
    const { asFragment } = render(<Button btnType={BUTTON_TYPES.ERROR}>Click me</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should call onClick when clicked', () => {
    const onButtonClick = jest.fn();

    render(<Button onClick={onButtonClick}>Click me</Button>);

    const button = screen.getByRole('button', {
      name: /click me/i,
    });

    userEvent.click(button);

    expect(onButtonClick).toBeCalledTimes(1);
  });
});
