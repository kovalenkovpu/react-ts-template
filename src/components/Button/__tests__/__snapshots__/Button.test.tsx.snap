// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL cwIfXK"
  >
    Click me
  </button>
</DocumentFragment>
`;
